core/testing.go

```diff
@@ -86,6 +88,7 @@ func StartTestNodeWithConfig(t *testing.T, cfg *TestConfig) testnode.Context {
 	t.Cleanup(func() {
 		err := cleanupCoreNode()
 		require.NoError(t, err)
+		require.NoError(t, removeDir(path.Join([]string{cfg.Tendermint.RootDir, "config"}...)))
```

Comment:
I think this should be part of the cleanupCoreNode func returned from `StartNode`, as it's clearly buggy. Think this way:
* You provide and API to setup something
* Additionally you provide API to clean up this something, as this something writes to disk
* Users calls this cleanup API, but it turns out to be not enough and users had to manually cleanup additional things.
* Do we want every user to stumble upon this themselves or do we want to do proper cleanup in one place without making users even noticing such issues?

Reply:
It is on app side:
https://github.com/celestiaorg/celestia-app/blob/b56c1bb292d8f4e16c4e55d47032551d23e2cf8e/testutil/testnode/full_node.go#L218

Except that it's for the default network, which, I guess, we expect the users to use if they don't wanna do any extra step.
It's a good idea to add it to `StartNode`. However, I added it here since you seem to depend on releases and not sure when next Celestia-app release will, and your CI will stay flaky until you update.

I will open a PR for this:
> I think this should be part of the cleanupCoreNode func returned from StartNode

And if you prefer to wait for this to come from next release, we can close this one :D

Reply:
PR on app side: https://github.com/celestiaorg/celestia-app/pull/1426

Reply:
Ok, it would have been nice to mention in the PR desc that this fix was merged into the app. The idea to have it temporarily here makes sense to me. Pls add the TODO here to remove once we update the app and we can merge this one

Reply:
[4faafcf](https://github.com/celestiaorg/celestia-node/pull/1822/commits/4faafcf19fcf837bc6945b2df1bb815f34b3b460)